nomad-clouddriver/src/main/kotlin/io/armory/plugin/nomad/JobCachingAgent.kt

```diff
@@ -62,6 +65,9 @@ class JobCachingAgent(val account: NomadCredentials) : CachingAgent, OnDemandAge
     override fun loadData(providerCache: ProviderCache?): CacheResult {
         val jobs = getJobs()
         val data = jobs.map {
+            // I'm not a fan of this. Spinnaker does far too much value conversions between maps and domain objects.
+            // What would it look like if we had a CacheData implementation that supported actual types, rather than
+            // forcing developers to convert back/forth between maps?
```

Comment:
Totally agree. I think the use of untyped mutable maps in Spinnaker is one of the worst issues it has. This is again, just because the current API expects a map, not my preference.
Stages has a similar issue with maps. One of the things that I like about SimpleStage is that it uses typed class instead of maps. I was hoping to add that into the real Stage API before we deprecate SimpleStage.